examples/jms-headers-and-properties/jms-headers-and-properties.bal

```diff
@@ -0,0 +1,63 @@
+import ballerina/jms;
+import ballerina/mb;
+import ballerina/log;
+
+// Initialize a JMS connection with the provider.
+jms:Connection conn = new ({
+    initialContextFactory: "wso2mbInitialContextFactory",
+    providerUrl: "amqp://admin:admin@carbon/carbon?brokerlist='tcp://localhost:5672'"
+});
+
+// Initialize a JMS session on top of the created connection.
+jms:Session jmsSession = new (conn, {
+    // An optional property that defaults to `AUTO_ACKNOWLEDGE`.
+    acknowledgementMode: "AUTO_ACKNOWLEDGE"
+});
+
+// Initialize a queue receiver using the created session.
+endpoint jms:QueueReceiver consumer {
+    session: jmsSession,
+    queueName: "MyQueue"
+};
+
+// Bind the created consumer to the listener service.
+service<jms:Consumer> jmsListener bind consumer {
+
+    // The `OnMessage` resource gets invoked when a message is received.
+    onMessage(endpoint consumer, jms:Message message) {
+      // Create a queue sender.
+        endpoint jms:SimpleQueueSender queueSender {
+            initialContextFactory: "wso2mbInitialContextFactory",
```

Comment:
Is it possible to have something like BallerinaInitialContextFactory?

Reply:
What about adding the **bmbInitialContextFactory**, since this is the **B**allerina **M**essage **B**roker initial context factory?  
Anyway we will keep the wso2mbInitialContextFactory as well to connect to WSO2 MB versions.